src/main/java/com/imaee/propinq/users/services/implementations/UserService.java

```diff
@@ -0,0 +1,177 @@
+        throwExceptionIfTokenIsExpired(activationTokenId);
+
+        User user = findUserToActivateOrThrowException(activationTokenId);
+        user.setActivated(true);
+        userRepository.save(user);
+
+        sendWelcomeEmail(user);
+
+    }
+    private void sendWelcomeEmail(User user) {
+        String emailBody = emailBuilder.buildWelcomeEmail(user);
+        emailService.sendEmail(user.getEmail(), "Welcome to PropInq", emailBody);
+    }
+
+    private void throwExceptionIfTokenIsExpired(UUID tokenId) {
+        if(isTokenExpired(tokenId)) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST,"Activation token has expired");
+        }
+    }
+
+    private boolean isTokenExpired(UUID tokenId) {
+        Token activationToken = findActivationTokenByIdOrThrowException(tokenId);
+        return activationToken.getTokenExpirationDate().isBefore(LocalDateTime.now());
+    }
+
+    private Token findActivationTokenByIdOrThrowException(UUID tokenId) {
+        return tokenService.findTokenByIdOrThrowException(tokenId);
+    }
+
+    private User findUserToActivateOrThrowException(UUID tokenId) {
+        User user = tokenService.findUserByTokenId(tokenId);
+        ifUserIsActivatedThrowException(user);
+        return user;
+    }
+
+    private void ifUserIsActivatedThrowException(User user) {
+        if(user.isActivated()){
+            throw new IllegalArgumentException("User is already activated.");
+        }
+    }
+
+    @Override
+    public void sendEmailToRecoverPassword(String email) {
+        User user = findUserByEmailOrThrowException(email);
+        Token recoverPasswordToken = tokenService.saveToken(user);
+        String emailContent = emailBuilder.buildRecoverPasswordEmail(user.getUsername(), recoverPasswordToken.getTokenId());
+
+        emailService.sendEmail(email, "Recover Password", emailContent);
+    }
+
+    private User findUserByEmailOrThrowException(String email) {
+        return userRepository.findByEmailAndDeletedIsFalse(email)
+                .orElseThrow(() -> new ResponseStatusException(HttpStatus.BAD_REQUEST, "User with email " + email +" does not exists."));
+    }
+
+    @Override
+    public void recoverPassword(RecoverPasswordRequest recoverPasswordRequest) {
+        throwExceptionIfPasswordAreNotEquals(recoverPasswordRequest.password(), recoverPasswordRequest.confirmPassword());
+        throwExceptionIfTokenIsExpired(recoverPasswordRequest.recoverPasswordToken());
+        User user = tokenService.findUserByTokenId(recoverPasswordRequest.recoverPasswordToken());
+        user.setPassword(passwordEncoder.encode(recoverPasswordRequest.password()));
+
+        userRepository.save(user);
+    }
+
+    public void throwExceptionIfPasswordAreNotEquals(String password, String confirmPassword) {
+        if(!password.equals(confirmPassword)) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Passwords do not match.");
+        }
+    }
```

Comment:
Malísimas las traducciones que usamos en megastore jajaja

```suggestion
    public void throwExceptionIfPasswordsDontMatch(String password, String confirmPassword) {
        if (!password.equals(confirmPassword))
            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Passwords do not match.");
    }
```